### DISCUSSION TITLE
Confusing `errors` shape on field array validation with `Yup.array()`

### DISCUSSION THREAD

**Discussion by stefanprobst** [react-hook-form/react-hook-form#3316] [Q&A]
hi, i have the following `yup` validation schema for a field array of the shape `people: Array<{ name: string }>`:

```js
const validationSchema = Yup.object({
  people: Yup.array()
    .of(Yup.object({ name: Yup.string().required() }))
    .required()
});
```

this will produce three different shapes of `errors.people` depending on whether it's (1) a field level error, (2) a fieldarray level error, or both (because the fieldarray level error does not seem to be cleared automatically):
1. an array `[{"name":{"message":"people[0].name is a required field","type":"required"}}]`
2. an object `{"message":"people is a required field","type":"required"}`
3. or an object-with-array-as-object: `{"0":{"name":{"message":"people[0].name is a required field","type":"required"}},"message":"people is a required field","type":"required"}`

this is very confusing - what is the recommended way of handling fieldarray errors?

note: [here](https://codesandbox.io/s/intelligent-elgamal-h9vng?file=/src/App.js) is a codesandbox to check the different error shapes

**Comment by bluebill1049** [comment]
what're the steps to reproduce different errors message?

rhf by default is at the field level to improve the perf of your form, instead of returning the entire form validation rules each keystroke.

**Reply by stefanprobst** [reply]
hi @bluebill1049 you can reproduce with the [CSB](https://codesandbox.io/s/intelligent-elgamal-h9vng?file=/src/App.js) linked above, when doing the following:

1. Click "Submit". => Error state is `{"message":"people is a required field","type":"required"}`
2. Click "Add". => Error state is still `{"message":"people is a required field","type":"required"}`
3. Type a character in the input, and delete it again. => Error state is `{"0":{"name":{"message":"people[0].name is a required field","type":"required"}},"message":"people is a required field","type":"required"}`

**Reply by bluebill1049** [reply]
1. Click "Submit". => Error state is `{"message":"people is a required field","type":"required"}` // there no input form level
2. Click "Add". => Error state is still `{"message":"people is a required field","type":"required"}` // the validation still failed with form level
3. Type a character in the input, and delete it again. => Error state is `{"0":{"name":{"message":"people[0].name is a required field","type":"required"}},"message":"people is a required field","type":"required"}` // now it's on the field level 

what is your expected error shape?

**Reply by stefanprobst** [reply]
> what is your expected error shape?

i'm not sure myself - maybe the confusion is just because the typescript types don't account for the fact that the field array level can have errors as well?

**Reply by miller-productions** [reply]
@stefanprobst , did you ever solve this?
